esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -0,0 +1,467 @@
+    log_provenance(caption, filename, cfg, provenance_info)
+
+    data_filename = get_diagnostic_filename(
+        f'weights_{variable}', cfg, extension='nc')
+    weights.to_netcdf(data_filename)
+
+
+def visualize_and_save_mean_weights(weights: dict,
+                                    cfg: dict,
+                                    provenance_info: list):
+    """Save the mean weights to a `.nc` file."""
+    weights_combined = (
+        xr.Dataset(weights)
+        .to_array(name='weight')
+        .mean(dim='variable')
+    )
+
+    weights_combined.attrs['short_name'] = 'weights'
+    weights_combined.attrs['units'] = 'a.u.'
+
+    label = 'combined mean'
+
+    filename = get_plot_filename('weights_combined', cfg)
+    caption = 'Mean weights for all variables'
+
+    barplot(weights_combined, label, filename)
+    log_provenance(caption, filename, cfg, provenance_info)
+
+    data_filename = get_diagnostic_filename(
+        'weights_combined', cfg, extension='nc')
+    weights_combined.to_netcdf(data_filename)
+
+
+def main(cfg):
+    """Perform climwip weighting method."""
+    observations = read_metadata(cfg, key='obs_data')
+    models = read_metadata(cfg, key='model_data')
+
+    variables = models.keys()
+
+    weights_dict = {}
+    ancestors = []
+
+    for variable in variables:
+
+        logger.info('Reading model data for %s', variable)
+        datasets_model = models[variable]
+        model_data, model_input_files = read_model_data(datasets_model)
+
+        logger.info('Reading observation data for %s', variable)
+        datasets_obs = observations[variable]
+        obs_data, obs_input_files = read_observation_data(datasets_obs)
+        obs_data = aggregate_obs_data(obs_data, operator='median')
+
+        logger.info('Calculating independence for %s', variable)
+        independence = calculate_independence(model_data)
+        visualize_and_save_independence(independence, cfg, model_input_files)
+        logger.debug(independence.values)
+
+        logger.info('Calculating performance for %s', variable)
+        performance = calculate_performance(model_data, obs_data)
+        visualize_and_save_performance(
+            performance, cfg, model_input_files + obs_input_files)
+        logger.debug(performance.values)
+
+        logger.info('Calculating weights for %s', variable)
+        sigma_performance = cfg['shape_params'][variable]['sigma_d']
+        sigma_independence = cfg['shape_params'][variable]['sigma_s']
+        weights = calculate_weights(
+            performance, independence,
+            sigma_performance, sigma_independence)
+        visualize_and_save_weights(
+            weights, cfg, model_input_files + obs_input_files)
+        logger.debug(weights.values)
+
+        weights_dict[variable] = weights
+        ancestors.extend(model_input_files)
+        ancestors.extend(obs_input_files)
+
+    visualize_and_save_mean_weights(
+        weights_dict, cfg, provenance_info=ancestors)
```

Comment:
As a follow up to the comment about renaming for readability: I would also replace the name `provenance_info` with `ancestor_files` or `input_files` everywhere in the script.